### PR TITLE
ditaa 0.10

### PR DESCRIPTION
**Pull request by apjanke**
Migrated from https://github.com/Homebrew/legacy-homebrew/pull/48372

### CODE REVIEW

**Comment by apjanke** [comment]
Install failed due to patch application failure.

```

==> Patching
patching file build/release.xml
Hunk #1 FAILED at 5.
1 out of 1 hunk FAILED -- saving rejects to file build/release.xml.rej
patching file src/org/stathissideris/ascii2image/core/CommandLineConverter.java
Hunk #1 FAILED at 46.
1 out of 1 hunk FAILED -- saving rejects to file src/org/stathissideris/ascii2image/core/CommandLineConverter.java.rej
Error: Failure while executing: /usr/bin/patch -g 0 -f -p1
```

**Comment by geoffholden** [comment]
I'll take a look at this and see why the patch is failing now.

**Comment by geoffholden** [comment]
I've submitted a pull request to your branch. The whitespace in the patch must've gotten messed up when it got moved to your branch.

**Comment by ilovezfs** [comment]
Maybe this should use `inreplace` instead of patching?

**Comment by mikemcquaid** [comment]
> Maybe this should use inreplace instead of patching?

Seems like a good idea 👍

**Comment by geoffholden** [comment]
I've updated the code on my pull request to @apjanke's branch to use inreplace.

**Comment by apjanke** [comment]
I pulled in @geoffholden's fix; it's working for me locally now.

File: Formula/ditaa.rb [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+  sha256 "82e49065d408cba8b323eea0b7f49899578336d566096c6eb6e2d0a28745d63b"
 
-  bottle :unneeded
+  depends_on :ant => :build
+  depends_on :java
 
   def install
-    libexec.install "ditaa0_9.jar"
-    bin.write_jar_script libexec/"ditaa0_9.jar", "ditaa"
+    # 0.10 Release didn't still calls itself 0.9
```
Comment (MikeMcQuaid) [inline Formula/ditaa.rb]: ` # 0.10 release still calls itself 0.9`

File: Formula/ditaa.rb [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+    # Reported upstream at https://github.com/stathissideris/ditaa/issues/14
+    inreplace "build/release.xml", "0_9", "0_10"
+    inreplace "src/org/stathissideris/ascii2image/core/CommandLineConverter.java", "ditaa version 0.9", "ditaa version 0.10"
+
+    mkdir "bin"
```
Comment (MikeMcQuaid) [inline Formula/ditaa.rb]: `bin.mkpath` is generally favoured but a tiny nit.

Comment (ilovezfs) [inline Formula/ditaa.rb]: That would actually create `"#{prefix}/bin"` which is a different directory. If you want to use `mkpath` here, it needs to be `(buildpath/"bin").mkpath`.

Comment (ilovezfs) [inline Formula/ditaa.rb]: But make he really does mean `bin.mkpath`?

Comment (MikeMcQuaid) [inline Formula/ditaa.rb]: I can't read, ignore me 😭

File: Formula/ditaa.rb (hunk 1 above)
Comment (apjanke) [inline Formula/ditaa.rb]: Fixed.

File: Formula/ditaa.rb (hunk 2 above)
Comment (apjanke) [inline Formula/ditaa.rb]: Yeah, it does need to be `bin` in the build dir, not target. Prevents a build error.

This should probably be looked in to and reported upstream as well.

**Comment by apjanke** [comment]
@BrewTestBot test this please

**Comment by MikeMcQuaid** [comment]
:shipit:

**Comment by apjanke** [comment]
Rebased on master due to `brew pull` failure.

```

$ brew pull --bottle --bump https://github.com/Homebrew/homebrew-core/pull/858
==> Fetching patch
Patch: https://github.com/Homebrew/homebrew-core/pull/858.patch
==> Applying patch
Applying: ditaa 0.10
Using index info to reconstruct a base tree...
M      	Formula/ditaa.rb
Falling back to patching base and 3-way merge...
Auto-merging Formula/ditaa.rb
CONFLICT (content): Merge conflict in Formula/ditaa.rb
error: Failed to merge in the changes.
Patch failed at 0001 ditaa 0.10
The copy of the patch that failed is found in: .git/rebase-apply/patch
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
Error: Patch failed to apply: aborted.
```

**Pull request closed without merging**